api/rs/slint/lib.rs

```diff
@@ -233,6 +233,14 @@ pub use i_slint_core::{
     string::{SharedString, ToSharedString},
 };
 
+/// Register a custom font from byte data at runtime.
+///
+/// Returns the number of font families that were registered from the provided data.
+#[cfg(feature = "shared-fontique")]
+pub fn register_font_from_memory(font_data: Vec<u8>) -> usize {
```

Comment:
I'm just wondering what's the most efficient and future proof API.
Also in the light of https://github.com/slint-ui/slint/issues/4250

I'm thinking we could have something like
`fn register_font_from_memory(font_data : impl AsRef[u8] + Sync + Send /* + 'static ? */) -> Result<???, RegisterFontError>`

That way we could also pass a `&'static str` embedded in the binary or in mapped memory.

I see that you are returning the count, any reason to do that?
Maybe we could return some kind of FontHandle? Not sure what the use-case is to get access to that.
However, I think this could error for several reason (invalid ttf, unsupported by the backend or renderer, and stuff like that.)